engines/sky/detection.cpp

```diff
@@ -201,6 +205,80 @@ DetectedGames SkyMetaEngine::detectGames(const Common::FSList &fslist) const {
 	return detectedGames;
 }
 
+Common::KeymapArray SkyMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace Sky;
+
+	Keymap *mainKeymap = new Keymap(Keymap::kKeymapTypeGame, "sky-main", "Beneath a Steel Sky");
+
+	Action *act;
+
+	act = new Action("LCLK", _("Walk / Look / Talk"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	mainKeymap->addAction(act);
+
+	act = new Action("RCLK", _("Use"));
+	act->setRightClickEvent();
+	act->addDefaultInputMapping("MOUSE_RIGHT");
+	act->addDefaultInputMapping("JOY_B");
+	mainKeymap->addAction(act);
+
+	act = new Action("CONFIRM", _("Confirm"));
+	act->setCustomEngineActionEvent(kSkyActionConfirm);
+	act->addDefaultInputMapping("RETURN");
+	act->addDefaultInputMapping("KP_ENTER");
+	mainKeymap->addAction(act);
+
+	act = new Action(kStandardActionSkip, _("Skip / Close"));
+	act->setCustomEngineActionEvent(kSkyActionSkip);
+	act->addDefaultInputMapping("ESCAPE");
+	act->addDefaultInputMapping("JOY_Y");
+	mainKeymap->addAction(act);
+
+	Keymap *shortcutsKeymap = new Keymap(Keymap::kKeymapTypeGame, SkyEngine::shortcutsKeymapId, "Beneath a Steel Sky - Shortcuts");
+
+	act = new Action(kStandardActionOpenMainMenu, _("Open control panel"));
+	act->setCustomEngineActionEvent(kSkyActionOpenControlPanel);
+	act->addDefaultInputMapping("F5");
+	act->addDefaultInputMapping("JOY_X");
+	shortcutsKeymap->addAction(act);
+
+	act = new Action("SKPL", _("Skip line"));
+	act->setCustomEngineActionEvent(kSkyActionSkipLine);
+	act->addDefaultInputMapping("PERIOD");
+	shortcutsKeymap->addAction(act);
+
+	act = new Action(kStandardActionPause, _("Pause"));
+	act->setCustomEngineActionEvent(kSkyActionPause);
+	act->addDefaultInputMapping("p");
+	shortcutsKeymap->addAction(act);
+
+	act = new Action("FAST", _("Toggle fast mode"));
+	act->setCustomEngineActionEvent(kSkyActionToggleFastMode);
+	act->addDefaultInputMapping("C+f");
+	shortcutsKeymap->addAction(act);
+
+	act = new Action("RFAST", _("Toggle really fast mode"));
+	act->setCustomEngineActionEvent(kSkyActionToggleReallyFastMode);
+	act->addDefaultInputMapping("C+g");
+	shortcutsKeymap->addAction(act);
+
+	act = new Action(kStandardActionOpenDebugger, _("Open debugger"));
```

Comment:
After looking at this. Could it possibly make a straightforward helper method, which would process the following:

```
keymap = {
  { kStandardActionOpenMainMenu,  _s("Open control panel"), kSkyActionOpenControlPanel, { "F5", "JOY_X", NULL } },
  { "RFAST", _s("Toggle really fast mode"), kSkyActionToggleReallyFastMode, { "C+g", NULL } },
 ...
};
```

Reply:
It would certainly be possible to make such a helper. However I did not choose this approach because:
* It makes the code harder to read. In the struct literal the names of the fields are not immediately visible.
* It makes the code harder to edit. If you want to change one of the action in a way that does not fit the struct / helper to experiment, you have to rewrite in to use the Action class directly anyway.